skipchain/api.go

```diff
@@ -40,146 +40,132 @@ func NewClient() *Client {
 	return &Client{Client: onet.NewClient("Skipchain")}
 }
 
-// NewLocalClient takes a LocalTest in order
-func NewLocalClient(local *onet.LocalTest) *Client {
-	return &Client{Client: local.NewClient("Skipchain")}
-}
-
-// CreateRootControl creates two Skipchains: a root SkipChain with
-// maximumHeight of maxHRoot and an control SkipChain with
-// maximumHeight of maxHControl. It connects both chains for later
-// reference.
-func (c *Client) CreateRootControl(elRoot, elControl *onet.Roster, baseHeight, maxHRoot, maxHControl int, ver VerifierID) (root, control *SkipBlock, cerr onet.ClientError) {
-	log.Lvl2("Creating root roster", elRoot)
-	root, cerr = c.CreateRoster(elRoot, baseHeight, maxHRoot, ver, nil)
-	if cerr != nil {
-		return
+// StoreSkipBlock asks the cothority to store the new skipblock, and eventually
+// attach it to the 'latest' skipblock.
+//  - latest is the skipblock where the new skipblock is appended. If el and d
+//   are nil, a new skipchain will be created with 'latest' as genesis-block.
+//  - el is the new roster for that block. If el is nil, the previous roster
+//   will be used.
+//  - d is the data for the new block. It can be nil. If it is not of type
+//   []byte, it will be marshalled using `network.Marshal`.
+func (c *Client) StoreSkipBlock(latest *SkipBlock, el *onet.Roster, d network.Message) (reply *StoreSkipBlockReply, cerr onet.ClientError) {
```

Comment:
`v2` doesn't have a `latest` anymore:

https://github.com/dedis/cothority/blob/skipchain_835_2/skipchain/msgs.go#L30

But it's very useful to have the `latest`, because it gives you the previous roster, so you know whom to contact:

https://github.com/dedis/cothority/blob/skipchain_835_2/skipchain/api.go#L140